### ISSUE TITLE
kraken tool kubectl get nodes fails due to missing computed_kubectl file.

### ISSUE THREAD

**Issue by davidewatson** [samsung-cnct/kraken#194]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Kraken version** (use `kraken version`):

rodin:plugins dwat$ kraken version
1.2.0-stable+git.sha.36456b1f383e7cbb74d2f739b51fc09a831fc420
rodin:plugins dwat$ 

**What happened**:

```
rodin:~ dwat$ kraken tool --config ~/.kraken/dew.yaml kubectl get nodes
Using Kraken config file: /Users/dwat/.kraken/dew.yaml 
Pulling image 'quay.io/samsung_cnct/k2:v0.5' ███████▒▒▒ Complete
WARNING: --output not specified. Using /Users/dwat/.kraken as location
WARNING: all not specified. Using 'all' as tags
ERROR! the playbook: get could not be found
cat: can't open '/tmp/computed_kubectl.fMBNME': No such file or directory
 
rodin:~ dwat$ 
```

**What you expected to happen**:

The command should run without error.

**How to reproduce it** (as minimally and precisely as possible):

Install the latest homebrew version of kraken and run the command above. Run `kraken generate` to ensure you have the latest config file.

**Anything else we need to know**:

This is a user-facing problem for the primary user interface of kraken-lib. Therefore I am making it a p0. We can downgrade it to a p1 if someone else can verify that they do not see the same behavior.

**Comment by davidewatson** [comment]
@DStorck: Verified that she also sees this behavior:

```
DStorck~/chart-logging (update-docs)$ kraken tool --config ~/.kraken/config.yaml  kubectl get nodes
Using Kraken config file: /Users/deirdrestorck/.kraken/config.yaml
Pulling image 'quay.io/samsung_cnct/k2:v0.5' █▒▒▒▒▒▒▒▒▒ Complete
WARNING: --output not specified. Using /Users/deirdrestorck/.kraken as location
WARNING: all not specified. Using 'all' as tags
ERROR! the playbook: get could not be found
cat: can't open '/tmp/computed_kubectl.LIikOb': No such file or directory
``` 

So it appears to be a real bug.

**Comment by alejandroEsc** [comment]
This is because k2 version has been upgraded past the changes i tried to make on https://github.com/samsung-cnct/kraken-lib/pull/854#pullrequestreview-69963963

which I was hoping would be updated now. Note that the change was to be backwards compatible to kraken-lib v0.2.0 at the time of the pr, please view

https://github.com/samsung-cnct/kraken/blob/44f39d239df24d55b2e22e5d9df0e73f06f76621/cmd/tool_kubectl.go#L36

so I will need to merge that PR mentioned above, AND have to create a new pr with the latest version of kraken-lib before the merge. @davidewatson @coffeepac

**Comment by davidewatson** [comment]
So the last working version of our k2 image (for kraken) was v0.2 which about a month old. I am going to triage our PR's since then and see if I can track down the culprit.

**Comment by davidewatson** [comment]
As a workaround you can use an older k2 image. This should only be down when running `kraken tool kubectl commands`, e.g.:

```
./kraken --image quay.io/samsung_cnct/k2:v0.2 --config ~/.kraken/dew.yaml tool kubectl get pods
```

**Comment by davidewatson** [comment]
So my first thought was to submit a PR changing the backwards compatibility check here: https://github.com/samsung-cnct/kraken/blob/44f39d239df24d55b2e22e5d9df0e73f06f76621/cmd/tool_kubectl.go#L36

However, with the subsequent commits to kraken I do not think this will suffice. Here is the current behavior:

From HEAD (44f39d239df24d55b2e22e5d9df0e73f06f76621)                                                                                           
We have:                                                                                                                                       
                                                                                                                                               
```                                                                                                                                            
rodin:kraken dwat$ make ; ./kraken --config ~/.kraken/dew.yaml tool kubectl -- get pods                                                        
Using Kraken config file: /Users/dwat/.kraken/dew.yaml                                                                                         
Pulling image 'quay.io/samsung_cnct/k2:latest' ... Complete                                                                                    
WARNING: --output not specified. Using /Users/dwat/.kraken as location                                                                         
WARNING: all not specified. Using 'all' as tags                                                                                                
ERROR! playbooks must be a list of plays                                                                                                       
                                                                                                                                               
The error appears to have been in '/Users/dwat/.kraken/dew.yaml': line 2, column 1, but may be elsewhere in the file depending on the exact syntax problem.                                                                                
                                                                                                                                               
The offending line appears to be:                                                                                                              
                                                                                                                                               
---                                                                                                                                            
version: v1                                                                                                                                    
^ here                                                                                                                                         
cat: can't open '/tmp/computed_kubectl.MeCbdf': No such file or directory                                                                      
                                                                                                                                               
rodin:kraken dwat$                                                                                                                             
```                                                                                                                                            
                                                                                                                                               
From the previous commit (9acd833cb089d1c3d3f1569faeaf59a46dec1cef)                                                                            
We have:                                                                                                                                       
                                                                                                                                               
```                                                                                                                                            
rodin:kraken dwat$ make ; ./kraken --config ~/.kraken/dew.yaml tool kubectl -- get pods                                                        
Using Kraken config file: /Users/dwat/.kraken/dew.yaml                                                                                         
Pulling image 'quay.io/samsung_cnct/k2:latest' ... Complete                                                                                    
WARNING: --output not specified. Using /Users/dwat/.kraken as location                                                                         
WARNING: all not specified. Using 'all' as tags                                                                                                
ERROR! the playbook: get could not be found                                                                                                    
cat: can't open '/tmp/computed_kubectl.LChICE': No such file or directory                                                                      
                                                                                                                                               
rodin:kraken dwat$                                                                                                                             
```

I am fine with merging the PR above, though there is one minor requested change.

@coffeepac: Thoughts? I don't want this to interfere with the etcd and k8s 1.8 merges, or conformance testing.

**Comment by davidewatson** [comment]
kraken 1.2.1 and kraken-lib v0.6 have been released. Once the above is merged, kraken will be brew installable. I believe this closes out this issue.

**Comment by coffeepac** [comment]
tested with kraken 1.2.1 and this works.
